### ISSUE TITLE
It doesn't work with input fields! - What should I do? [Update: it does!]

### ISSUE THREAD

**Issue by justinwyllie** [iansinnott/react-string-replace#78]
I'm doing this:
```
<div>
            
            {reactStringReplace('abc___def', /(___)/g, (match, i) => (
                <input key={i} >{match}</input>
            ))} 
</div>
```
I get an error in the console which doesn't give any more information. Prior to the error I get the warning: "Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function."

It is the input field which is the problem - it works with a span.

The container function is a functional component.

**Comment by iansinnott** [comment]
The error sounds strange, but in general `input`s do not have children. Try `<input value={match} />`.

**Comment by justinwyllie** [comment]
> The error sounds strange, but in general `input`s do not have children. Try `<input value={match} />`.

Hi Ian
Thanks a lot. That was it. I was just about reaching the same conclusion myself by manually rendering the array. But - you probably spotted it at a glance. That is the error. As you say a rather a strange error message. When I rendered an array manually I got the correct error message about input fields and no children. It even supports: onChange handler:

```
let html = reactStringReplace('abc___def', /(___)/g, (match, i) => (
        <input value={match} key={i} onChange={handleChange} />
  ));
```

Thanks again.
